ide/example_code/initial-bot.rs

```diff
@@ -39,27 +38,27 @@ pub fn init(api: &mut api::Api) -> bool
  * - boost (bool):    When set to true, your snake will get a speed boost, but
  *                    this will cost some of the snake’s mass.
  */
-pub fn step(api: &mut api::Api) -> (bool, f32, bool)
-{
-	// let's start by moving in a large circle. Please note that all angles are
-	// represented in radians, where -π to +π is a full circle.
-	let angle: f32 = 0.001;
+pub fn step(api: &mut api::Api) -> (bool, f32, bool) {
+    // let's start by moving in a large circle. Please note that all angles are
+    // represented in radians, where -π to +π is a full circle.
+    let angle: f32 = 0.001;
 
-	// check for other snakes
-	let segments = api.get_segments();
+    // check for other snakes
+    let segments = api.get_segments();
 
-	for seg in segments {
-		if !seg.is_self && seg.dist < 20.0 {
-			// you can send log messages to your browser or any other viewer with the
-			// appropriate Viewer Key.
-			api.log("Oh no, I'm going to die!");
-			break;
-		}
-	}
+    for seg in segments {
+        if !seg.is_self && seg.dist < 20.0 {
+            // you can send log messages to your browser or any other viewer with the
+            // appropriate Viewer Key. It's okay for the call to fail, e.g. when the
+            // rate limit is reached
```

Comment:
Just a small comment: As I said in the [gameserver pull request](https://github.com/schlangenprogrammiernacht/gameserver/pull/8#issuecomment-1356919509), the bot cannot be informed about being rate limited, so this example is misleading. I'd suggest changing it to "e.g. when the log memory is full".

Reply:
Sorry, totally my fault. Fixed it in both PRs